File: examples/jump/unit_commitment/unit_commitment_test.jl
import DotEnv
DotEnv.config()

using MOSInterface

@testset "Unit Commitment" begin

    # Interface
    interface = Interface()

    # Delete model
    delete_model_with_name(interface, "Stochastic Unit Commitment")

    # New model
    model = new_model(interface, "./examples/jump/unit_commitment/unit_commitment_model.jl")

    # Get model
    model = get_model_with_name(interface, "Stochastic Unit Commitment")

    # Set inputs
    set_interface_object(model, "n_g", 10)
    set_interface_object(model, "n_gsl", 5)

    # Show info
    @test get_system(model) == "jump"

    # Solve
    MOSInterface.run(model)

    # Status
    @test get_status(model) == "success"

end

File: src/jump/unit_commitment/unit_commitment_test.py
from dotenv import load_dotenv, find_dotenv
load_dotenv(find_dotenv())

from mos.interface import Interface

def test_unit_commitment():

    # Interface
    interface = Interface()

    # Delte model
    interface.delete_model_with_name('Stochastic Unit Commitment')

    # New model
    model = interface.new_model('./examples/jump/unit_commitment/unit_commitment_model.jl')

    # Existing model by name
    model = interface.get_model_with_name('Stochastic Unit Commitment')

    # Set inputs
    model.set_interface_object('n_g', 10)
    model.set_interface_object('n_gsl', 5)

    # Solve 
    model.run()

    assert(model.get_status() == 'success')

    print(model.get_execution_log())

if __name__ == '__main__':
    test_unit_commitment()
